client/apps/edit/components/content/sections/image_collection/test/controls.test.js

```diff
@@ -1,154 +1,151 @@
       )
-      component.instance().addArtworkFromUrl([
-        {type:'image', url:'image.com'},
-        {type: 'artwork', image:'artwork.jpg'}
-      ])
-      expect(component.instance().props.section.get('images')[0].url).toMatch('image.com')
-      expect(component.instance().props.section.get('images')[1].image).toMatch('artwork.jpg')
+      expect(props.section.get('images')[3].image).toMatch('artwork.jpg')
     })
 
     it('#inputsAreDisabled returns false if layout is not fillwidth', () => {
-      const component = mount(
-        <Controls {...props} />
-      )
+      const component = getWrapper(props)
+
       const inputsAreDisabled = component.instance().inputsAreDisabled(props.section)
       expect(inputsAreDisabled).toBe(false)
     })
 
     it('#inputsAreDisabled returns true if fillwidth section has images', () => {
       props.section.set({layout: 'fillwidth'})
-      const component = mount(
-        <Controls {...props} />
-      )
+      const component = getWrapper(props)
+
       const inputsAreDisabled = component.instance().inputsAreDisabled(props.section)
       expect(inputsAreDisabled).toBe(true)
     })
 
-    it('Calls #fillwidthAlert if inputsAreDisabled and trying to add an image', () => {
-      const component = mount(
-        <Controls {...props} />
-      )
-      component.find('.edit-controls__artwork-inputs').simulate('click')
-      expect(Controls.prototype.fillwidthAlert.called).toBe(true)
+    it('Calls #logErrorAction if inputsAreDisabled and trying to add an image', () => {
+      props.section.set({layout: 'fillwidth'})
+      const component = getWrapper(props)
+
+      component.find('.edit-controls__artwork-inputs').at(0).simulate('click')
+      expect(component.props().logErrorAction.mock.calls.length).toBe(1)
     })
   })
 
   describe('Image Set inputs', () => {
-    it('Renders the active image_set layout', () => {
+    beforeEach(() => {
       props.section.set({type: 'image_set', layout: 'mini'})
-      const component = shallow(
-        <Controls {...props} />
-      )
+    })
+
+    it('Renders the active image_set layout', () => {
+      const component = getWrapper(props)
+
       expect(component.html()).toMatch(
         '<div class="input-group"><div class="radio-input" data-active="true"></div>Mini</div>'
       )
     })
 
     it('changes imageset layout on button click', () => {
-      props.section.set({type: 'image_set', layout: 'mini'})
-      const component = shallow(
-        <Controls {...props} />
-      )
+      const component = getWrapper(props)
+
       component.find('.radio-input').at(1).simulate('click')
       expect(props.section.get('layout')).toMatch('full')
     })
 
     it('changes imageset title on input', () => {
-      const component = mount(
-        <Controls {...props} />
-      )
+      const component = getWrapper(props)
       const input = component.find('.edit-controls__image-set-inputs').find('input')
+
```

Comment:
Nice tests, but there's some pretty new functionality in controls that would be worthwhile to cover in tests. For example, `fetchDenormalizedArtwork`, `filterAutocomplete` and `fillwidthAlert`. I would consider adding at least one for each of them (would be a nice follow up for that `toLowerCase`!).